contracts/WhitelistRegistry.sol

```diff
@@ -51,11 +55,27 @@ contract WhitelistRegistry is Ownable {
     }
 
     function setWhitelistLimit(uint256 whitelistLimit_) external onlyOwner {
+        whitelistLimitNew = whitelistLimit_;
+        if (whitelistLimitNew >= _whitelist.length()) {
+            whitelistLimit = whitelistLimitNew;
+        }
+        emit WhitelistLimitSet(whitelistLimit_);
+    }
+
+    function shrinkWhitelist(uint256 partition) external {
         uint256 whitelistLength = _whitelist.length();
-        if (whitelistLimit_ < whitelistLength) {
-            _shrinkPoorest(_whitelist, whitelistLength - whitelistLimit_);
+        if (whitelistLength == 0) revert EmptytWhitelist();
+        if (whitelistLimitNew >= whitelistLength) revert NoShrinkInWhitelist();
+        address[] memory addresses = _whitelist.items.get();
+        unchecked {
+            for (int256 i = int256(whitelistLength) - 1; i >= 0; --i) {
```

Comment:
Instead of conversions, maybe better to do while loop and break when i == 0 after checking balance condition?

Reply:
why do we need to iterate from the end?

Reply:
We change the object which we iterate, and when we iterate from start and we remove some elements we have to change loop boundary (or we need recalculate `whitelistLength` each iteration) and change `i`, because deleted position is filling by last element in object